bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/ArithmeticGroupFunction.java

```diff
@@ -537,27 +542,29 @@ public Threshold(State upperValue, State lowerValue, DecimalType upperLimit, Dec
         }
 
         @Override
-        public State calculate(Set<Item> items) {
-
-            if (items != null && items.size() > 0) {
-                boolean isLower = true;
-                for (Item item : items) {
-                    DecimalType itemState = (DecimalType) item.getStateAs(DecimalType.class);
-                    if (upperLimit.compareTo(itemState) < 0) {
-                        return upperState;
-                    } else if (lowerLimit.compareTo(itemState) <= 0) {
-                        isLower = false;
-                    }
-                }
-                if(isLower) {
-                    return lowerState;
-                }
-                return null;
-            } else {
-                // if we do not have any items, we return the passive state
-                return lowerState;
-            }
-        }
+		public State calculate(Set<Item> items) {
+			if (items != null && items.size() > 0) {
+				boolean isLower = true;
+				for (Item item : items) {
+					DecimalType itemState = (DecimalType) item.getStateAs(DecimalType.class);
+					if (getFunctype(item).equals("ft_value_1_byte")) {

```

Comment:
I think I’d like it better if we pushed the magic constant string down into the method.

A _concrete_ version of the method could return a boolean, and we’d call it like

```
if (usesPercentType(item)) {
    itemState = itemState.as(PercentType.class);
}
```

A more _generic_ version could instead return the State to use, like

```
Class<? extends DecimalType> type = findItemType(item);
DecimalType itemState = item.getStateAs(type);
```

(this would also replace the line above)